test/cmd/diff.sh

```diff
@@ -41,6 +44,11 @@ run_kubectl_diff_tests() {
     output_message=$(kubectl diff -f hack/testdata/pod-changed.yaml || test $? -eq 1)
     kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.0'
 
+    # Test found diff with server-side apply
+    kubectl apply -f hack/testdata/pod.yaml
+    output_message=$(kubectl diff -f hack/testdata/pod-changed.yaml --server-side --force-conflicts || test $? -eq 1)
```

Comment:
I wasn't expecting to find the `test $? -eq 1`, I'm not sure what you're trying to do here?

Reply:
Oh this is for a diff between `hack/testdata/pod.yaml` (CSA) and `hack/testdata/pod-changed.yaml` (SSA).

This exercises `kubectl diff --server-side` to validate that the original issue is fixed and `kubectl diff --server-side` is usuable since now we pass fieldManager

Reply:
I think it's confusing that I added `kubectl apply -f hack/testdata/pod.yaml` first

Reply:
You're doing this in a subshell, does it propagate and cause a failure when triggered?

Reply:
Yeah, the test fails even if the error is in the sub-shell